### PR TITLE
[giflib] Update to 5.2.1

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?  
  - [x] Updates giflib to version 5.2.1.
  - [x] Modernize portfile.
  - [x] Disable `GifDrawBoxedText8x8` on Windows (needs `strtok_r`).
  - [x] Enables shared linkage.
    (Apart from general usefulness, it resolves libtool trouble for mingw-dynamic triplets e.g. when building gdal.) 

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  yes

### CODE REVIEW

**Comment by dg0yt** [comment]
Hm, so Windows has no `strtok_r`. But why doesn't it fail in static builds? If no one ever calls `GifDrawBoxedText8x8`, I better disable this function instead of adding a patch which I cannot test. (Windows' `strtok_s` might work as a replacement.) Opinions?

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM, thanks for your updated. @dg0yt

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

File: ports/giflib/fix-compile-error.patch [hunk 1]
```diff
@@ -73,4 +73,39 @@ index f30b61b..18dc4a2 100644
  
      GifFile->Private = (void *)Private;
      Private->FileHandle = FileHandle;
-
+diff --git a/gif_font.c b/gif_font.c
```
[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/giflib/fix-compile-error.patch:76]: There is also another `strtok_s` in newer C standards. I'm unable to build and test the result, and I'm not aware that anyone links to it.

File: ports/giflib/exports.def [hunk 2]
```diff
@@ -0,0 +1,55 @@
+EXPORTS
```
[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/giflib/exports.def:1]: Upstream is: giflib. Friendly put, upstream is not very active. Last commit is from 2019-08-17. Release 5.1.2 is from 2019-06-24. TIckets and merge requests are without feedback. (I almost forgot that I submitted a CMake build system.) Last not least `build.adoc` states that "Requests to go any further out of the way to accommodate
Microsoft's toolchains are unlikely be looked on favorably." 

On the positive side, it means there won't be a large burden from maintaining the DEF file.

**Review by vicroms — APPROVED** [review]

**Pull request merged**